test/secret/secret_test.go

```diff
@@ -0,0 +1,191 @@
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			// Modify Secret key with new value
+			secretKey := "hec_token"
+			modifiedKey := "6659B1AE-09B5-7101-530D-DA3FFD929501"
+			testenv.ModifySecretKey(deployment, testenvInstance.GetName(), secretKey, modifiedKey)
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Wait for Standalone to be in READY status
+			testenv.StandaloneReady(deployment, deployment.GetName(), standalone, testenvInstance)
+
+			//Once system is up after update check each pod for secret key update
+			standaloneSecretName := fmt.Sprintf(testenv.SecretObjectStandalonePodName, deployment.GetName(), 2)
+			licenseMasterSecretName := fmt.Sprintf(testenv.SecretObjectLicenseMasterName, deployment.GetName(), 2)
+			monitoringConsoleSecretName := fmt.Sprintf(testenv.SecretObjectMonitoringConsoleName, testenvInstance.GetName(), 2)
+			verificationPods := []string{standaloneSecretName, licenseMasterSecretName, monitoringConsoleSecretName}
+
+			//Verify that each pod has the updated secret object
+			updated := testenv.VerifySecretObjectUpdate(deployment, testenvInstance, verificationPods, secretKey, modifiedKey)
+			Expect(updated).To(Equal(true), "All keys not updated")
+
+		})
+	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("secret: Secret update on indexers and search head cluster", func() {
+
+			// Download License File
+			licenseFilePath, err := testenv.DownloadFromS3Bucket()
+			Expect(err).To(Succeed(), "Unable to download license file")
+
+			// Create License Config Map
+			testenvInstance.CreateLicenseConfigMap(licenseFilePath)
+
+			err = deployment.DeploySingleSiteCluster(deployment.GetName(), 3, true /*shc*/)
+			Expect(err).To(Succeed(), "Unable to deploy cluster")
+
+			// Ensure that the cluster-master goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Modify Secret key with new value
+			secretKey := "hec_token"
+			modifiedKey := "6659B1AE-09B5-7101-530D-DA3FFD929501"
+			testenv.ModifySecretKey(deployment, testenvInstance.GetName(), secretKey, modifiedKey)
+
+			// Ensure that the cluster-master goes to Ready phase
+			//testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure indexers go to Ready phase
+			testenv.SingleSiteIndexersReady(deployment, testenvInstance)
+
+			// Ensure search head cluster go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify MC Pod is Ready
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Verify RF SF is met
+			testenv.VerifyRFSFMet(deployment, testenvInstance)
```

Comment:
Nit: Just curious, are these blocked calls?

Reply:
They are effective checks to see that the pods accept the new updated credentials and comes up in ready state. i.e no issues with new secret object details